src/layer/x86/instancenorm_x86.cpp

```diff
@@ -0,0 +1,217 @@
+// Tencent is pleased to support the open source community by making ncnn available.
+//
+// Copyright (C) 2025 THL A29 Limited, a Tencent company. All rights reserved.
+//
+// Licensed under the BSD 3-Clause License (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// https://opensource.org/licenses/BSD-3-Clause
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#include "instancenorm_x86.h"
+
+#if __SSE2__
+#include <emmintrin.h>
+#if __AVX__
+#include <immintrin.h>
+#endif
+#endif // __SSE2__
+
+namespace ncnn {
+
+int InstanceNorm_x86::forward_inplace(Mat& bottom_top_blob, const Option& opt) const
+{
+    // x = (x - mean) / (sqrt(var + eps)) * gamma + beta
+
+    int w = bottom_top_blob.w;
+    int h = bottom_top_blob.h;
+    int c = bottom_top_blob.c;
+    int size = w * h;
+
+    #pragma omp parallel for num_threads(opt.num_threads)
+    for (int q = 0; q < c; q++)
+    {
+        float* ptr = bottom_top_blob.channel(q);
+
+        // mean and var
+        float sum = 0.f;
+        float sqsum = 0.f;
+
+        int i = 0;
+#if __SSE2__
+#if __AVX__
+#if __AVX512F__
+        __m512 _sum_avx512 = _mm512_setzero_ps();
+        for (; i <= size - 16; i += 16)
+        {
+            __m512 _p = _mm512_loadu_ps(ptr + i);
+            _sum_avx512 = _mm512_add_ps(_sum_avx512, _p);
+        }
+        sum = _mm512_reduce_add_ps(_sum_avx512);
+#endif // __AVX512F__
+        __m256 _sum_avx = _mm256_setzero_ps();
+        for (; i <= size - 8; i += 8)
+        {
+            __m256 _p = _mm256_loadu_ps(ptr + i);
+            _sum_avx = _mm256_add_ps(_sum_avx, _p);
+        }
+
+        __m128 vlow = _mm256_castps256_ps128(_sum_avx);
+        __m128 vhigh = _mm256_extractf128_ps(_sum_avx, 1);
+        __m128 hsum_128 = _mm_add_ps(vlow, vhigh);
+        hsum_128 = _mm_hadd_ps(hsum_128, hsum_128);
+        hsum_128 = _mm_hadd_ps(hsum_128, hsum_128);
+        sum += _mm_cvtss_f32(hsum_128);
```

Comment:
use `_mm_reduce_add_ps` `_mm256_reduce_add_ps` `_mm512_comp_reduce_add_ps` in `x86_usability.h`